Add additional requirements/options for OPs and RPs regarding re-auth…

gffletch opened the pull request:
…entication

Addresses issue #92

Added mandatory requirements to the OP requirements 

Outlined options for the RP in driving the desired behavior for re-authentication

gffletch committed:
Add additional requirements/options for OPs and RPs regarding re-authentication

Addresses issue #92

Added mandatory requirements to the OP requirements
Outlined options for the RP in driving the desired behavior for re-authentication

seanmillerrsa commented on draft-openid-ipsie-sl1-profile.md:
```diff
@@ -165,7 +165,9 @@ For the authorization code flow, OpenID Providers:
 * MUST support `nonce` parameter values up to 64 characters in length, and MAY reject `nonce` values longer than 64 characters.
-* MUST support the `max_age` parameter with a values representing the maximum number of seconds allowable since the user was authenticated by the OP. If the elapsed time since authentication is less than this value, the OP MAY choose to actively reauthenticate the user.  If the elapsed time since authentication is greater than this value, the OP MUST actively reauthenticate the user.
+* MUST support `prompt=login` where if specified, the OP MUST explicitly (re)authenticate the user using a user visible challenge. If the user is currently logged in as determined by the OP, then the OP MUST only re-authenticate that user.
+* MUST support the `max_age` parameter with a values representing the maximum number of seconds allowable since the user was authenticated by the OP. If the elapsed time since authentication is less than the specified value OR the OP can silently authenticate the user and issue a new `auth_time` within the required `max_age`, then no user visible authentication is required. The OP MAY choose to e ...
```
did you mean 'value' instead of 'values'?  Rest reads great

aaronpk replied:
```suggestion
* MUST support the `max_age` parameter with a value representing the maximum number of seconds allowable since the user was authenticated by the OP. If the elapsed time since authentication is less than the specified value OR the OP can silently authenticate the user and issue a new `auth_time` within the required `max_age`, then no user visible authentication is required. The OP MAY choose to explicitly (re)authenticate the user regardless.
```

seanmillerrsa commented on draft-openid-ipsie-sl1-profile.md:
```diff
@@ -195,7 +197,9 @@ For the authorization code flow, Relying Parties:
 * SHOULD NOT use `nonce` parameter values longer than 64 characters;
+* SHOULD use `prompt=login` if the RP wants to ensure that the user is visibly challenged for (re)authentication
 * SHOULD use the `max_age` parameter in the authentication request to specify the maximum allowable authentication age to the OP in seconds.  The value of the `max_age` parameter MAY be determined based upon the business rules of the RP.
+* SHOULD user `max_age` and `prompt=none` if the RP wants to ensure that the user is not visibly challenged for (re)authentication.
```
to match the prompt=login requirement above, suggest we change the order in this sentence to "SHOULD use `prompt=none` and `max_age`"

Note the typo right now user instead of use

aaronpk committed:
Update draft-openid-ipsie-sl1-profile.md

gffletch committed:
Merge branch 'openid:main' into re-auth

gffletch committed:
Fix typo in max_age and prompt recommendation

Addressed comments in the PR. Also a note that once we land on terminology, we may want to update this document.
